# 4.3.1.
1. Usando  uma  representação  gráfica adequada,  compare  as  distribuições  do  número  de mortes 

prematuras (Value) em **Portugal**, **Espanha, França e Itália.** 

## *Imports* estáticos

In [68]:
import pandas as pd


## Leitura de Ficheiros

In [69]:
dados = pd.read_csv('../dados/AIRPOL_data.csv', delimiter=";", header=0)
dados = dados.drop(columns=['Unnamed: 8', 'Unnamed: 9', 'Unnamed: 10', 'Unnamed: 11', 'Unnamed: 12', 'Unnamed: 13', 'Unnamed: 14', 'Unnamed: 15'])

### Classificação das Variáveis
4 Variáveis:



**X**: Valores Médios de Poluente PM2.5 em Portugal: Tipos de Variavel contínua.
**Y**: Valores Médios de Poluente PM2.5 em Espanha:  Tipos de Variavel contínua.
**Z**: Valores Médios de Poluente PM2.5 em França:   Tipos de Variavel contínua.
**T**: Valores Médios de Poluente PM2.5 em Italia:   Tipos de Variavel contínua.


# Filteração de dados.

In [70]:
filtered_pm25 = dados[(dados['Air_Pollutant'] == 'PM2.5') & (dados['Country'].isin(['Portugal', 'France', 'Spain', 'Italy']))]
print(filtered_pm25[['Country', 'Air_Pollution_Average[ug/m3]']])

      Country Air_Pollution_Average[ug/m3]
1141   France                         6,20
1148   France                         4,50
1149   France                         5,20
1157   France                         4,50
1158   France                         4,50
...       ...                          ...
47743   Italy                        11,70
47763   Italy                        11,70
47780   Italy                        11,70
47787   Italy                        11,70
47792   Italy                        11,70

[6714 rows x 2 columns]


# Encontrar primeiro País com menor numero de amostras

In [71]:
min_length = min(len(filtered_pm25[filtered_pm25['Country'] == 'Portugal']), len(filtered_pm25[filtered_pm25['Country'] == 'France']), len(filtered_pm25[filtered_pm25['Country'] == 'Spain']), len(filtered_pm25[filtered_pm25['Country'] == 'Italy']))

# Filtro por Cada País.

In [72]:
# Filter data for each country
portugal = filtered_pm25[filtered_pm25['Country'] == 'Portugal']['Air_Pollution_Average[ug/m3]'].str.replace(',', '.').astype(float)
spain = filtered_pm25[filtered_pm25['Country'] == 'Spain']['Air_Pollution_Average[ug/m3]'].str.replace(',', '.').astype(float)
france = filtered_pm25[filtered_pm25['Country'] == 'France']['Air_Pollution_Average[ug/m3]'].str.replace(',', '.').astype(float)
italy = filtered_pm25[filtered_pm25['Country'] == 'Italy']['Air_Pollution_Average[ug/m3]'].str.replace(',', '.').astype(float)


# Tabela de Correlação

In [73]:

# Sample data to equalize lengths
sampled_portugal = portugal.sample(n=min_length, random_state=42).reset_index(drop=True)
sampled_spain = spain.sample(n=min_length, random_state=42).reset_index(drop=True)
sampled_france = france.sample(n=min_length, random_state=42).reset_index(drop=True)
sampled_italy = italy.sample(n=min_length, random_state=42).reset_index(drop=True)

# Combine sampled data into a single DataFrame
df_corr_sampled = pd.concat([sampled_portugal, sampled_spain, sampled_france, sampled_italy], axis=1)
df_corr_sampled.columns = ['Portugal', 'Spain', 'France', 'Italy']

correlation_sampled = df_corr_sampled.corr().round(3)
print(correlation_sampled)



          Portugal  Spain  France  Italy
Portugal     1.000 -0.058  -0.040  0.001
Spain       -0.058  1.000  -0.002  0.000
France      -0.040 -0.002   1.000 -0.036
Italy        0.001  0.000  -0.036  1.000


## Conclusão da Matriz de Correlação

A matriz evidencia correlações **muito fracas** entre os países no que diz respeito aos níveis médios de poluição por PM2.5:

- A **maior correlação** observada é entre **Portugal e Espanha** (r = -0.058), ainda assim considerada **muito fraca** e negativa.
- As restantes correlações encontram-se muito próximas de zero, o que sugere que **não existe relação linear significativa** entre os níveis de poluição dos diferentes países.
- Estes resultados indicam que os valores de PM2.5 **variam de forma independente** entre **Portugal**, **Espanha**, **França** e **Itália**.


# Testes - Nenhum:
- Nenhum teste é pedido a fazer no enunciado e , não existe qualquer teste para apropriado nos slides para testar a correlação entre os paíse ( Pearson, Spearman ou Kendal).
Não foi visto sentido de testar algo desta forma,como específicado a cima.
